client/otagent/src/main/java/org/hawkular/apm/otagent/OpenTracingManager.java

```diff
@@ -0,0 +1,439 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.apm.otagent;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Stack;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.hawkular.apm.api.logging.Logger;
+import org.hawkular.apm.api.logging.Logger.Level;
+import org.hawkular.apm.api.utils.PropertyUtil;
+import org.jboss.byteman.rule.Rule;
+import org.jboss.byteman.rule.helper.Helper;
+
+import io.opentracing.Span;
+import io.opentracing.SpanContext;
+import io.opentracing.Tracer;
+import io.opentracing.Tracer.SpanBuilder;
+import io.opentracing.propagation.Format;
+import io.opentracing.propagation.TextMap;
+
+/**
+ * This class provides helper capabilities to the byteman rules.
+ *
+ * @author gbrown
+ */
+public class OpenTracingManager extends Helper {
+
+    private static final Logger log = Logger.getLogger(OpenTracingManager.class.getName());
+
+    private Tracer tracer = OpenTracingTracer.getSingleton();
+
+    public static final Format<TextMap> TEXT_MAP = io.opentracing.propagation.Format.Builtin.TEXT_MAP;
```

Comment:
Is this needed? On line 85 you can directly return this constant from opentracing